From 8b1669a3f21a20467d6344dd6db0dcf2af1d2383 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Fri, 27 Jul 2012 23:20:47 +0800
Subject: libmediaplayerservice: fix the problem that streaming exist when change to low bandwidth

BZ: 48710

Httplive Bandwidth change: When change to low, the ts stream became only audio stream.
google's orignal design does not cover this case and will pop "can not play", try to
cover the case.

orig-patch:
35716:When change to low, the tsstream became only audio stream exists

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: I00ec373ef50b2471ca3ee81611ebda3cb8c22918
Orig-Change-Id: Icffbe3c9200a994a13cc0abfb729d41387a17d2a
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../nuplayer/NuPlayerRenderer.cpp                  |    4 ++--
 media/libstagefright/ACodec.cpp                    |    3 ++-
 media/libstagefright/mpeg2ts/ATSParser.cpp         |    4 ++--
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 942633c..f3e7fd7 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -581,7 +581,7 @@ void NuPlayer::Renderer::onFlush(const sp<AMessage> &msg) {
 
         Mutex::Autolock autoLock(mFlushLock);
         mFlushingAudio = false;
-
+        mHasAudio = false;
         mDrainAudioQueuePending = false;
         ++mAudioQueueGeneration;
 
@@ -591,7 +591,7 @@ void NuPlayer::Renderer::onFlush(const sp<AMessage> &msg) {
 
         Mutex::Autolock autoLock(mFlushLock);
         mFlushingVideo = false;
-
+        mHasVideo = false;
         mDrainVideoQueuePending = false;
         ++mVideoQueueGeneration;
 
diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 11de21f..e70893a 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -3816,7 +3816,8 @@ bool ACodec::LoadedState::onConfigureComponent(
 
     sp<RefBase> obj;
     if (msg->findObject("native-window", &obj)
-            && strncmp("OMX.google.", mCodec->mComponentName.c_str(), 11)) {
+            && strncmp("OMX.google.", mCodec->mComponentName.c_str(), 11)
+            && !strncasecmp(mime.c_str(), "video/", 6)) {
         sp<NativeWindowWrapper> nativeWindow(
                 static_cast<NativeWindowWrapper *>(obj.get()));
         CHECK(nativeWindow != NULL);
diff --git a/media/libstagefright/mpeg2ts/ATSParser.cpp b/media/libstagefright/mpeg2ts/ATSParser.cpp
index 175a263..3fb39e1 100644
--- a/media/libstagefright/mpeg2ts/ATSParser.cpp
+++ b/media/libstagefright/mpeg2ts/ATSParser.cpp
@@ -396,8 +396,8 @@ status_t ATSParser::Program::parseProgramMap(ABitReader *br) {
         }
 
         if (!success) {
-            ALOGI("Stream PIDs changed and we cannot recover.");
-            return ERROR_MALFORMED;
+            mStreams.clear();
+            ALOGI("Stream PIDs changed and we try to recover.");
         }
     }
 
-- 
1.7.4.1

